File: app/src/main/java/com/temperoni/recipes/component/RecipesApplicationComponent.java
package com.temperoni.recipes.component;

import android.content.Context;

import com.temperoni.recipes.RecipesApplication;
import com.temperoni.recipes.ui.activities.RecipesListActivity;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author Leandro Temperoni
 */

@Singleton
@Component(modules = {
        ApplicationModule.class,
        RecipesModule.class})
public interface RecipesApplicationComponent {

    RecipesComponent getRecipesComponent();
}

File: app/src/main/java/com/temperoni/recipes/component/RecipesApplicationComponent.kt
package com.temperoni.recipes.component

import dagger.Component
import javax.inject.Singleton

/**
 * @author Leandro Temperoni
 */
@Singleton
@Component(modules = [(ApplicationModule::class), (RecipesModule::class)])
interface RecipesApplicationComponent {

    val recipesComponent: RecipesComponent
}
